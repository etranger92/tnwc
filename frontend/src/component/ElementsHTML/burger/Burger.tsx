import React from "react";
import "./burger.css";

//clickEvent
type Props = {
  // eslint-disable-next-line @typescript-eslint/ban-types
  clickEvent: Function;
  isItClicked: boolean;
};
const Burger: React.FC<Props> = (props) => {
  // eslint-disable-next-line react/prop-types
  const { clickEvent, isItClicked } = props;

  return (
    <div
      data-testid="no-classes"
      id="burger"
      onClick={() => clickEvent()}
      className={isItClicked ? "cross" : ""}
    >
      <span></span>
      <span></span>
      <span></span>
    </div>
  );
};

export default Burger;
