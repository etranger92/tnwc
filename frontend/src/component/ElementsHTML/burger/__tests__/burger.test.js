import * as React from 'react';
import Burger from "../Burger.tsx"
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe("We test the innerHTML of the burger", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Burger />);
  });
  test("Snapshot burger ", () => {
    //It works
 expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("burger style css", () => {
    //it doesn't work
  wrapper.setProps({ isItClicked: false });
  expect(wrapper.find("#burger").getDOMNode()).not.toHaveStyle('position: absolute')
  }); 
  test("burger mutes to a cross by adding 'cross class' ", () => {
  wrapper.setProps({ isItClicked: true });
  expect(wrapper.find("#burger").getDOMNode()).toHaveClass("cross");
  }); 
  });
