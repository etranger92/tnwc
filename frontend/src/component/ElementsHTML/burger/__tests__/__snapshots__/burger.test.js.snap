// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`We test the innerHTML of the burger Snapshot burger  1`] = `
<Burger>
  <div
    className=""
    data-testid="no-classes"
    id="burger"
    onClick={[Function]}
  >
    <span />
    <span />
    <span />
  </div>
</Burger>
`;
